src/tsickle_compiler_host.ts

```diff
@@ -0,0 +1,199 @@
+import * as ts from 'typescript';
+
+import {convertDecorators} from './decorator-annotator';
+import {processES5} from './es5processor';
+import {ModulesManifest} from './modules_manifest';
+import {annotate} from './tsickle';
+
+export enum Pass {
+  None,
+  DecoratorDownlevel,
+  Tsickle
+}
+
+export interface TsickleCompilerHostOptions {
+  googmodule: boolean;
+  es5Mode: boolean;
+  tsickleTyped: boolean;
+  prelude: string;
+}
+
+export interface TsickleEnvironment {
+  /**
+   * If true, tsickle and decorator downlevel processing will be skipped for
+   * that file.
+   */
+  shouldSkipTsickleProcessing: (fileName: string) => boolean;
+  /** Converts a path to a module name */
```

Comment:
This description is obvious from the function.  Instead can you say why someone would define it, or the sorts of transformations?  I think this does e.g. "foo/bar" => "foo.bar".  What does the context param do?